### PR TITLE
Use accounting_match in test_array_sub_new_res_in_hook

### PR DESCRIPTION
**Pull request by Shrini-h**
#### Bug/feature Description
* *PTL Test case TestRunJobHook.test_array_sub_new_res_in_hook can fail if run using non root account*

#### Affected Platform(s)
* *All*

#### Cause / Analysis / Design
* *Tracejob_match needs root permission to match in Accounting logs*

#### Solution Description
* *use accountint_match instead*

#### Testing logs/output
[Test_report_logs.zip](https://github.com/PBSPro/pbspro/files/2935542/Test_report_logs.zip)

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: test/tests/functional/pbs_runjob_hook.py [hunk 1]
```diff
@@ -110,8 +110,8 @@ def test_array_sub_new_res_in_hook(self):
             sid = j1.create_subjob_id(jid, i)
             m = "'runjob_hook' hook set job's Resource_List.site = site_value"
             self.server.tracejob_match(m, id=sid, n='ALL', tail=False)
-            m = "Resource_List.site=site_value"
-            self.server.tracejob_match(m, id=sid, n='ALL', tail=False)
+            m = 'E;' + re.escape(sid) + '.*Resource_List.site=site_value'
+            self.server.accounting_match(m, regexp=True, max_attempts=3)
```
Comment (anamikau) [inline test/tests/functional/pbs_runjob_hook.py]: please remove max_attempts=3 or update it to 20.

**Comment by Shrini-h** [comment]
Rebased and squashed. Thank you @bhroam and @anamikau  for signoff

**Comment by Shrini-h** [comment]
PR ready for merge, @PBSPro/Maintainers please take it forward

**Comment by bhroam** [comment]
Thanks @Shrini-h I merged in the PR

**Pull request merged**